apps/emqx_gateway/src/coap/emqx_coap_session.erl

```diff
@@ -162,7 +162,7 @@ deliver(Delivers, Ctx, Session) ->
           end,
     lists:foldl(Fun,
                 #{out => [], session => Session},
-                Delivers).
+                lists:reverse(Delivers)).
```

Comment:
We should reverse the `Out` list instead of  `Delivers`.

Otherwise, messages in front of `Delivers` are given a larger seq-id or msg-id.

Reply:
it ok, CoAP message id not mean it order